src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2041,13 +2041,13 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
             BinaryOperatorKind op = binary.OperatorKind.Operator();
 
-            // learn from non-null constant
+            // learn from non-null expression
```

Comment:
>learn from non-null expression [](start = 15, length = 30)

I didn't understand the change in comment. It looks like we're still learning if there is a non-null constant (`binary.Right.ConstantValue?.IsNull != true`)